cli/azd/cmd/auto_install.go

```diff
@@ -453,3 +468,67 @@ func ExecuteWithAutoInstall(ctx context.Context, rootContainer *ioc.NestedContai
 	// Normal execution path - either no args, no matching extension, or user declined install
 	return rootCmd.ExecuteContext(ctx)
 }
+
+// CreateGlobalFlagSet creates a new flag set with all global flags defined.
+// This is the single source of truth for global flag definitions.
+func CreateGlobalFlagSet() *pflag.FlagSet {
+	globalFlags := pflag.NewFlagSet("global", pflag.ContinueOnError)
+
+	globalFlags.StringP("cwd", "C", "", "Sets the current working directory.")
+	globalFlags.Bool("debug", false, "Enables debugging and diagnostics logging.")
+	globalFlags.Bool(
+		"no-prompt",
```

Comment:
Can we avoid the duplication here?  We should not re-define commands or flags as part of auto-install..  All flags are already registered in the root cobra command